src/scrot.c

```diff
@@ -570,21 +570,14 @@ static Imlib_Image scrotGrabShot(void)
 static void scrotExecApp(Imlib_Image image, struct tm *tm, char *filenameIM,
     char *filenameThumb)
 {
-    char *execStr;
-    int ret;
-
-    execStr = imPrintf(opt.exec, tm, filenameIM, filenameThumb, image);
-
-    errno = 0;
-
-    ret = system(execStr);
+    char *execStr = imPrintf(opt.exec, tm, filenameIM, filenameThumb, image);
+    int ret = system(execStr);
 
     if (ret == -1)
         warn("The child process could not be created");
-    else if (WEXITSTATUS(ret) == 127)
+    else if (WIFEXITED(ret) && WEXITSTATUS(ret) == 127)
```

Comment:
This is bit of a band-aid/ad-hoc fix. If the child didn't exit normally, we'd probably want to report that as well.